src/NFSFile.cpp

```diff
@@ -209,8 +209,8 @@ ssize_t CNFSFile::Write(void* context, const void* lpBuf, size_t uiBufSize)
     //danger - something went wrong
     if (writtenBytes < 0)
     {
-      kodi::Log(ADDON_LOG_ERROR, "Failed to pwrite(%s) %s\n",
-                ctx->filename.c_str(), nfs_get_error(ctx->pNfsContext));
```

Comment:
This .c_str() needs to stay. printf-like functions require a char* pointer and will crash if given a c++ object.

Also do NFS urls have usernames/passwords in them? Should the filename be redacted in logs?